File: DSA/Aarohi-Sharma/Day33_q2.cpp
// User function Template for C++
class Solution {
  public:
    string smallestNumber(int s, int d) {
        // code here
        int st=pow(10,d-1);
        int end=pow(10,d)-1;
        
        while(st <= end){
            
            int sum=0;
            int num=st;
            
            while(num>0){
                
                sum+=num%10;
                num/=10;
            }
            
            if(sum==s) return to_string(st);
            
            st++;
            
        }
        
        return "-1";
    }
};

File: DSA/Yashvardhan-Guliyani/Day33_q2.java
class Solution {
    public String smallestNumber(int s, int d) {
        if(s<1 || s>d*9)
            return "-1";
        String str="";
        int sum=s;
        int i=1;
        while(i<=9){
            if(sum-i<=(d-1)*9){
                str+=i;
                d--;
                sum=sum-i;
                i=-1;
            }
            if(d==0)
                break;
            i++;
        }
        return str;
    }
}
